auto_round/mllm/mllm_dataset.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+from typing import Dict
+
+
+import torch
+from torch.utils.data import Dataset, DataLoader
+from PIL import Image
+from transformers.data.data_collator import default_data_collator
+
+from .utils import _extract_data_dir
+from .template import Template, TEMPLATES, load_template
+
+
+MLLM_DATASET : Dict[str, Dataset] = {}
+
+def register_dataset(name):
+    def register(dataset):
+        MLLM_DATASET[name] = dataset
+        return dataset
+    return register
+
+
+@register_dataset("llava")
+class LlavaDataset(Dataset):
+    """Dataset for supervised fine-tuning."""
+
+    def __init__(
+            self,
+            model_type_or_template,
+            model,
+            tokenzier,
+            dataset_path,
+            extra_data_dir,
+            max_length,
+            padding=True,
+            truncation=True,
+            ) -> None:
+        super().__init__()
```

Comment:
Please enhance the code by adding some comments and better add a README to include instructions on how to prepare llava dataset , support new models, etc. in this folder. Additionally, better test the code with an unseen model to evaluate its generalization capabilities.